Sources/StreamChatUI/Navigation/ChatChannelListRouter.swift

```diff
@@ -0,0 +1,41 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+/// A `NavigationRouter` subclass that handles navigation actions of `ChatChannelListVC`.
+public typealias ChatChannelListRouter = _ChatChannelListRouter<NoExtraData>
+
+/// A `NavigationRouter` subclass that handles navigation actions of `ChatChannelListVC`.
+open class _ChatChannelListRouter<ExtraData: ExtraDataTypes>:
+    NavigationRouter<_ChatChannelListVC<ExtraData>>,
+    ComponentsProvider
+{
+    /// Presents the view controller with the profile of the current user.
+    open func showCurrentUserProfile() {
+        debugPrint("Not implemented", #function)
+    }
+
+    /// Presents the view controller with messages for the provided cid.
```

Comment:
nit: `Presents..` wording confused me here 🤔 Can we maybe say `show` to follow the method naming patter?

Reply:
Rebased to apply changes from #1086 and fixed